packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -186,7 +169,16 @@ export default class GraphQLRunner {
     const document = this.parse(query)
     const errors = this.validate(schema, document)
 
-    const result =
+    let tracer
+    if (parentSpan) {
+      const tracer = new GraphQLSpanTracer(`GraphQL Query ${queryName}`, {
```

Comment:
This is only working because the `query` is called _after_ initialization of the whole file, not because classes are "hoisted", because they are not.

I would reconsider doing it this way...